### PR TITLE
Update nuget & msbuild versions

### PR DESCRIPTION
**Pull request by wtgodbe**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -204,8 +204,8 @@
     <MicrosoftWebWebView2Version>1.0.705.50</MicrosoftWebWebView2Version>
     <MicrosoftWebXdtVersion>1.4.0</MicrosoftWebXdtVersion>
     <SystemIdentityModelTokensJwtVersion>6.8.0</SystemIdentityModelTokensJwtVersion>
-    <NuGetVersioningVersion>5.9.0-rc.7121</NuGetVersioningVersion>
-    <NuGetFrameworksVersion>5.9.0-rc.7121</NuGetFrameworksVersion>
+    <NuGetVersioningVersion>5.9.0</NuGetVersioningVersion>
+    <NuGetFrameworksVersion>5.9.0</NuGetFrameworksVersion>
```
Comment (dougbu) [inline eng/Versions.props:208]: nit: You're using 5.8.0 for NuGet packages in #31035. Should we be consistent❔

**Review by dougbu — APPROVED** [review]

File: eng/Versions.props (hunk 1 above)
Comment (wtgodbe) [reply eng/Versions.props:208]: Done

**Comment by wtgodbe** [comment]
Tests like `Microsoft.Extensions.ApiDescription.Client.GetOpenApiReferenceMetadataTest.Execute_SetsClassName_BasedOnOutputPath` are consistently failing in Helix with:

> Assert.Equal() Failure
Expected: SortedDictionary<String, String> [[ClassName, ThisIsClassy], [CodeGenerator, NSwagCSharp], [FirstForGenerator, true], [Namespace, Console.Client], [OriginalItemSpec, TestProjects/files/NSwag.json], ...]
Actual:   SortedDictionary<String, String> [[ClassName, ThisIsClassy], [CodeGenerator, NSwagCSharp], [FirstForGenerator, true], [Namespace, Console.Client], [OriginalItemSpec, TestProjects/files/NSwag.json], ...]

From this file: https://github.com/dotnet/aspnetcore/blob/main/src/Tools/Extensions.ApiDescription.Client/test/GetOpenApiReferenceMetadataTest.cs. It's not obvious to me how these could be related to nuget/msbuild updates - @dougbu looks like you were the last to touch those tests, any ideas?

**Comment by dougbu** [comment]
@wtgodbe the problem looks more like an encoding change that could easily be related to the updated MSBuild dependencies. Looking&hellip;

**Comment by dougbu** [comment]
My bet is some metadata is now skipped in results from MSBuild methods. In addition, comparison failures like that seen in `SerializeMetadata_ReturnsExpectedString()` may indicate an ordering change as well. But, the full strings and collections aren't visible in the Helix logs.

Do you already have a local build of this branch❔ If yes, debug the failing tests to see what the general issue is and adjust the test expectations. Again, `SerializeMetadata_ReturnsExpectedString()` is probably a special case.

**Comment by wtgodbe** [comment]
@dougbu looks like with the new msbuild, some metadata is getting serialized in a different order - see https://github.com/dotnet/aspnetcore/pull/31064/commits/757656783e9017f36c5f6ef649ea021b308e3811 for how it changed. `SerializeMetadata_ReturnsExpectedString` has the same root cause as the other tests, as they're all comparing against expected metadata where the elements use the old ordering. If this seems reasonable/expected to you I can fix up the other tests as well

**Comment by dougbu** [comment]
@wtgodbe the changes in 757656783e9017f36c5f6ef649ea021b308e3811 look fine but I am unsure the other tests are seeing the same problem because they use `new SortedDictionary<string, string>(StringComparer.Ordinal)` and the related collection comparisons are still failing. Go ahead w/ your fixes while I scratch my head and frown 😺

**Comment by wtgodbe** [comment]
Yes, they use a `SortedDictionary`, but that just sorts the elements by their keys - the last element still has key `SerializedMetadata`, but the `SerializedMetadata` value is just the string that now looks different than it used to.

**Comment by wtgodbe** [comment]
Tests are working for me locally now, will merge once this is green.

**Comment by dougbu** [comment]
> the last element still has key `SerializedMetadata`

Thanks❕ That was the piece I missed.

**Pull request merged**